front_end/content.py

```diff
@@ -855,16 +855,20 @@ def save_exercise_score(self, course_id, assignment_id, exercise_id, user_id, ne
 
     def get_submissions_basic(self, course_id, assignment_id, exercise_id, user_id):
         submissions = []
-        sql = '''SELECT submission_id, date, passed
-                 FROM submissions
+        sql = '''SELECT s.submission_id, s.date, s.passed, s.partner_id,
+                 CASE WHEN s.partner_id NOT NULL
```

Comment:
I like your creativity, but this approach is still not optimal because it has to do a separate SELECT statement for every time there is a non-null partner_id value. I better way (I didn't think of this earlier) would be to do a right join rather than an inner join. An inner join only retains rows that have non-null values on the join columns. But a right (or left) join keeps them as null. See [this article](https://www.geeksforgeeks.org/sql-join-set-1-inner-left-right-and-full-joins/).

As a side note, I recommend that you download this software: https://sqlitebrowser.org. You can copy the database to your computer and then run SQL commands on the database as you are testing different options.

Let me know if you're not able to figure out the joins still, and we can look at it together.

Reply:
Oh okay, that makes sense! Wow, I've never used left or right joins before, I'm liking SQL more all the time. Thanks for the article and explanation!